x/ibc/09-localhost/client_state.go

```diff
@@ -0,0 +1,270 @@
+package localhost
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	connectionexported "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/exported"
+	channelexported "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/exported"
+	commitmentexported "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/exported"
+	commitmenttypes "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/types"
+	ibctypes "github.com/cosmos/cosmos-sdk/x/ibc/types"
+)
+
+var _ clientexported.ClientState = ClientState{}
+
+// ClientState requires (read-only) access to keys outside the client prefix.
+type ClientState struct {
+	store types.KVStore
+}
+
+// NewClientState creates a new ClientState instance
+func NewClientState(store types.KVStore) ClientState {
+	return ClientState{
+		store: store,
+	}
+}
+
+// GetID returns the loop-back client state identifier.
+func (cs ClientState) GetID() string {
+	return clientexported.Localhost.String()
+}
+
+// GetChainID returns an empty string
+func (cs ClientState) GetChainID() string {
+	return ""
```

Comment:
Hmm, I don't know about this, maybe this should be `ctx.ChainID()`? Maybe non-essential though

Reply:
@cwgoes I think we need to add a `ctx sdk.Context` to the `ClientState` field. 

```go
// ClientState requires (read-only) access to keys outside the client prefix.
type ClientState struct {
	ctx   sdk.Context
	store sdk.KVStore
}
```

@gregdhill can you add it and add this following function too?:

```go
// WithContext updates the client state context to provide the chain ID and latest height
func (cs *ClientState) WithContext(ctx sdk.Context) {
    cs.ctx = ctx
}
```

Reply:
```suggestion
	return cs.ctx.ChainID()
```